### ISSUE TITLE
Small Bug: Collections do not reflect later renaming of grains

### ISSUE THREAD

**Issue by nicolas-arnold** [sandstorm-io/sandstorm#3282]
Hey there,

if I add some Grains to a collection, and then rename the Grain the new name does not get reflected in the collection.

My mental model of the collections is, they are simply pointers to existing documents, and therefore should be notified or simply automatically refer to the actual name of the Grain it is pointing to, instead of memorizing the initial name.

If the initial name is considered important, it could still be available on hover or writting in parentheses, but if you add a "Untitled" Grain and then want to rename it, or if a document changes heavily over time and thus needs a new name, the current behaviour is not the best, I think.

Thank in advance for your thoughts on that!

**Comment by ocdtrekkie** [comment]
I suspect this is because apps don't really have a way to tell other apps their name has changed. Collections looks like a part of Sandstorm, but it is  "just an app" and doesn't have easy access to that information.

It may be good to make an issue on the Collections app's repo as well, though I will tag this feature request here as well, since I am not sure if it would require functionality from Sandstorm to do.

**Comment by zenhack** [comment]
Yeah, I'd have to dig a bit but I imagine this isn't trivial to implement for the reasons @ocdtrekkie states (though I agree it would be a good improvement). I suspect we will need modifications to sandstorm core, so probably makes sense to keep the issue open here as well.

**Comment by nicolas-arnold** [comment]
Thanks for the quick response and positive feedback. Is it easiest if I copy this to the Collections App as well, or do you have a better procedure (or think it's not necessary yet)?

I can see, how even a document's name can be something that can contain confidential information and should not generally be accessible. As the whole document is to be accessed within Collection, it should not be an issue here. 

So the Collections-App only gets initial access to the Grain and behind the scenes just manages the propagation of access rights for the user with access to the Collection?

**Comment by ocdtrekkie** [comment]
If you leave an issue on the Collections app as well, it's a good reference if people work on it specifically.

I'm not sure the exact method the Collections app gets grain titles and what is required for it to re-get them. It seems sensible it should be able to check them when someone opens the Collection to view it.

**Comment by kentonv** [comment]
At the time that you add a grain to a collection, I think the title of the grain is communicated via the `PowerboxTag`. I'm not sure if there is currently an API for the collection app to ask for the current title. We could add one. But, note that it may be confusing for a grain in a collection to change titles. The collection app should probably do something similar to what Sandstorm's grain list UI does with shared grains whose titles have changed: show both the old and new titles.
